Microsoft.Toolkit.Uwp.UI.Controls/Menu/MenuItem.cs

```diff
@@ -0,0 +1,152 @@
+            get { return (string)GetValue(HeaderProperty); }
+            set { SetValue(HeaderProperty, value); }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MenuItem"/> class.
+        /// </summary>
+        public MenuItem()
+        {
+            DefaultStyleKey = typeof(MenuItem);
+            IsFocusEngagementEnabled = true;
+        }
+
+        /// <summary>
+        /// This method is used to show the menu for current item
+        /// </summary>
+        public void ShowMenu()
+        {
+            FlyoutButton?.Flyout?.ShowAt(FlyoutButton);
+        }
+
+        /// <summary>
+        /// This method is used to hide the menu for current item
+        /// </summary>
+        public void HideMenu()
+        {
+            FlyoutButton?.Flyout?.Hide();
+        }
+
+        /// <inheritdoc />
+        protected override void OnApplyTemplate()
+        {
+            FlyoutButton = GetTemplateChild("FlyoutButton") as Button;
+            IsOpened = false;
+
+            if (FlyoutButton != null && Items != null && Items.Any())
+            {
+                var menuFlyout = new MenuFlyout();
+                menuFlyout.Placement = ParentMenu.Orientation == Orientation.Horizontal
+                    ? FlyoutPlacementMode.Bottom
+                        : FlyoutPlacementMode.Right;
+
+                menuFlyout.Opened -= MenuFlyout_Opened;
+                menuFlyout.Closed -= MenuFlyout_Closed;
+                FlyoutButton.PointerExited -= FlyoutButton_PointerExited;
+                menuFlyout.Opened += MenuFlyout_Opened;
+                menuFlyout.Closed += MenuFlyout_Closed;
+                FlyoutButton.PointerExited += FlyoutButton_PointerExited;
+
+                menuFlyout.MenuFlyoutPresenterStyle = ParentMenu.MenuFlyoutStyle;
+                FlyoutButton.Style = ParentMenu.HeaderButtonStyle;
+
+                foreach (var item in Items)
+                {
+                    var menuItem = item as MenuFlyoutItemBase;
+                    if (menuItem != null)
+                    {
+                        menuFlyout.Items.Add(menuItem);
+                    }
+                    else
+                    {
+                        var newMenuItem = new MenuFlyoutItem();
+                        newMenuItem.DataContext = item;
+                        menuFlyout.Items.Add(newMenuItem);
+                    }
+                }
+
+                FlyoutButton.Flyout = menuFlyout;
+            }
+
+            base.OnApplyTemplate();
+        }
+
+        private void FlyoutButton_PointerExited(object sender, PointerRoutedEventArgs e)
+        {
+            if (IsOpened)
+            {
+                VisualStateManager.GoToState(FlyoutButton, "Opened", true);
+            }
+        }
+
+        private void MenuFlyout_Closed(object sender, object e)
+        {
+            IsOpened = false;
+            VisualStateManager.GoToState(FlyoutButton, "Normal", true);
```

Comment:
Just wondering on the VisualStates, are they there for the reason that someone might want to retemplate the control and add visual states? If so, that should be documented in the docs. I didn't see Visual States in the default template.